plugins/index.ts

```diff
@@ -179,23 +179,44 @@ extendEnvironment((hre) => {
       hre.log(
         'Found existing proxy at:',
         proxyAddress,
-        ' attempting to upgrade instance',
+        'attempting to upgrade instance',
         contractName
       );
+      const existingImplementationAddress =
+        await hre.upgrades.erc1967.getImplementationAddress(proxyAddress!);
```

Comment:
we should just wrap this whole thing in an if statement above so we don't have to use non-null assertions everywhere. I think if it's null it's going to throw `EIP1967BeaconNotFound` errors anyways so might as well catch that higher up before we start setting memos